### DISCUSSION TITLE
Docker versioning custom tags

### DISCUSSION THREAD

**Discussion by wrighbr** [renovatebot/renovate#25659] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.42.0

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Cloud (bitbucket.org)

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I have a docker images with a custom tag `custom-go-image:0.43.2-go1.20.8` I want the tag to bumped like below

| current tag | new tag |
| ---  | --- |
| 0.43.2-go1.20.8 | 0.49.0-go1.21.1 |

### What you tried so far.

```json
 {
   "packageRules": [
        {
            "matchDatasources": [
                "docker"
            ],
            "matchDepNames": [
                "custom.docker.register.com/custom-go-image"
            ],
            "versioning": "regex:^(?<major>\\d+)(\\.?<minor>\\d+)?(\\.(?<patch>\\d+))?(-go1(?<build>\\d+))(?(\\.?<revision>\\d+)?$"
        }
}
```

### Relevant debug logs

<details><summary>Logs</summary>

```
DEBUG: Dependency custom.docker.register.com/custom-go-image has unsupported/unversioned value 0.43.2-go1.20.8 (versioning=regex:^(?<major>\d+)(\.?<minor>\d+)(\.?<patch>\d+)-go.(\.?<build>\d+)(\.?<revision>\d+)?$) (repository=orgname/hello-world)
```

</details>

**Comment by rarkins** [comment]
Try loose versioning if you're not concerned about the go version part

**Reply by wrighbr** [reply]
Im interested in the go part

**Reply by rarkins** [reply]
Please describe in plain english how Renovate should decide whether to upgrade or not, and how to sort between releases

**Reply by wrighbr** [reply]
It would be great to have two sets of rules. One set would increase the version number in the tag, for example, changing from `0.43.2-go1.20.8` to `0.49.0-go1.20.8`. The other set would specifically update the Go version, like changing from `0.43.2-go1.20.8` to `0.43.2-go1.21.1`.

**Reply by rarkins** [reply]
That's why I asked if you're interested in the go part.

**Reply by wrighbr** [reply]
I have been playing around with package rules I'm not sure if it's possibly to have 2 package rules for the same dependency.

See example below. The first rule updates the will update `0.43.2` -> `0.49.0` and leave the go version alone. The second rule will update both the first part of the tag and go version.

```
    "packageRules": [
        {
            "matchDatasources": [
                "docker"
            ],
            "matchDepNames": [
                "custom.docker.register.com/custom-go-image"
            ],
            "versioning": "regex:^(?<major>\\d+)(\\.(?<minor>\\d+))?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$"
        },
        {
            "matchDatasources": [
                "docker"
            ],
            "matchDepNames": [
                "custom.docker.register.com/custom-go-image"
            ],
            "versioning": "regex:^.*-go(?<major>\\d+)\\.(?<minor>\\d+)?(\\.(?<patch>\\d+))?(-(?<compatibility>.*))?$"
        }
    ]
```

However the last rule in the array seems to alway win. Is it possible to have 2 package rules for the same dependency?

**Reply by rarkins** [reply]
No, what you need to do is have dependencies. Create a custom regex manager rule to get the second
